src/components/shared/AddUserDialogue.tsx

```diff
@@ -0,0 +1,93 @@
+import React, { useState } from 'react';
+
+// material ui
+import DialogTitle from '@material-ui/core/DialogTitle';
+import Dialog from '@material-ui/core/Dialog';
+import CheckIcon from '@material-ui/icons/Check';
+import ClearIcon from '@material-ui/icons/Clear';
+import { green, red, amber } from '@material-ui/core/colors';
+import List from '@material-ui/core/List';
+import ListItem from '@material-ui/core/ListItem';
+import ListItemIcon from '@material-ui/core/ListItemIcon';
+import ListSubheader from '@material-ui/core/ListSubheader';
+import ListItemText from '@material-ui/core/ListItemText';
+import TextField from '@material-ui/core/TextField';
+import { fade, makeStyles, withStyles, Theme, createStyles } from '@material-ui/core/styles';
+
+export interface AddUserProps {
+    title: string;
+    open: boolean;
+    selectedValue: boolean;
+    onClose: (value: boolean, firstname: string, lastname: string, email: string, church: string) => void;
+}
+
+export function AddUserDialog(props: AddUserProps) {
+  const classes = useStyles();
+  const [firstName, setFirstName] = useState<string>('');
+  const [lastName, setLastName] = useState<string>('');
+  const [email, setEmail] = useState<string>('');
+  const [church, setChurch] = useState<string>('');
+
+  let { onClose, selectedValue, open, title } = props;
+
+  const handleClose = () => {
+    onClose(selectedValue, firstName, lastName, email, church);
+  };
+
+  const handleOptionClick = (value: boolean) => {
+    onClose(value, firstName, lastName, email, church);
+  };
+  
+  const handleFirstNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setFirstName(event.target.value);
+  }
+
+  const handleLastNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setLastName(event.target.value);
+  }
+
+  const handleEmailChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setEmail(event.target.value);
+  }
+
+  const handleChurchChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setChurch(event.target.value);
+  }
+
+  return (
+    <Dialog onClose={handleClose} open={open}>
+      <DialogTitle id='confirm-dialog'>{title} </DialogTitle>
+      <form className={classes.root} noValidate autoComplete="off">
+        <TextField onChange={handleFirstNameChange} id="firstname" label="firstname" /><br/>
```

Comment:
it's pretty normal to do the setChange functions in the onChange prop, instead of making a handleChange function for each

`onChange ={(event) => setChange(event.target.value)}`